openedx/adg/lms/webinar/models.py

```diff
@@ -0,0 +1,103 @@
+"""
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.core.validators import FileExtensionValidator
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from model_utils.models import TimeStampedModel
+
+from openedx.adg.lms.applications.helpers import validate_file_size
+
+from .constants import ALLOWED_BANNER_EXTENSIONS, BANNER_MAX_SIZE
+
+
+class Webinar(TimeStampedModel):
+    """
+    Model to create a webinar
+    """
+
+    start_time = models.DateTimeField(verbose_name=_('Start Time'), )
+    end_time = models.DateTimeField(verbose_name=_('End Time'), )
+
+    title = models.CharField(verbose_name=_('Title'), max_length=255, )
+    description = models.TextField(verbose_name=_('Description'), )
+    presenter = models.ForeignKey(
+        User, verbose_name=_('Presenter'), on_delete=models.CASCADE, related_name='webinar_presenter',
+    )
+    meeting_link = models.URLField(default='', verbose_name=_('Meeting Link'), blank=True, )
+    banner = models.ImageField(
+        upload_to='webinar/banners/',
+        verbose_name=_('Banner'),
+        validators=[FileExtensionValidator(ALLOWED_BANNER_EXTENSIONS)],
+        help_text=_('Accepted extensions: .png, .jpg, .jpeg, .svg'),
+    )
+
+    language = models.CharField(verbose_name=_('Language'), choices=settings.LANGUAGES, max_length=2,)
+
+    co_hosts = models.ManyToManyField(User, verbose_name=_('Co-Hosts'), blank=True, related_name='webinar_co_hosts', )
+    panelists = models.ManyToManyField(
+        User, verbose_name=_('Panelists'), blank=True, related_name='webinar_panelists',
+    )
+
+    UPCOMING = 'upcoming'
+    DELIVERED = 'delivered'
+    CANCELLED = 'cancelled'
+
+    STATUS_CHOICES = (
+        (UPCOMING, _('Upcoming')),
+        (DELIVERED, _('Delivered')),
+        (CANCELLED, _('Cancelled'))
+    )
+    status = models.CharField(
+        verbose_name=_('Webinar Status'), choices=STATUS_CHOICES, max_length=10, default=UPCOMING,
+    )
+
+    is_virtual = models.BooleanField(default=True, verbose_name=_('Virtual Event'), )
+    created_by = models.ForeignKey(
+        User, verbose_name=_('Created By'), on_delete=models.CASCADE, blank=True, related_name='webinar_created_by',
+    )
+    modified_by = models.ForeignKey(
+        User,
+        verbose_name=_('Last Modified By'),
+        on_delete=models.CASCADE,
+        null=True,
+        blank=True,
+        related_name='webinar_modified_by',
+    )
+
+    class Meta:
+        app_label = 'webinar'
+
+    def clean(self):
+        if self.start_time > self.end_time:
+            raise ValidationError({'start_time': _('End date must be greater than start date')})
+        if self.banner:
+            error = validate_file_size(self.banner, BANNER_MAX_SIZE)
+            if error:
+                raise ValidationError({'banner': error})
+
+    def __str__(self):
+        return self.title
```

Comment:
Let's change the ordering here. Move `__str__ `above `clean()` I have read it [here](https://steelkiwi.com/blog/best-practices-working-django-models-python/)